apps/cde-ui/src/app/pages/create-visualization-page/create-visualization-page.component.ts

```diff
@@ -0,0 +1,169 @@
+import { CommonModule } from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  EventEmitter,
+  inject,
+  OnInit,
+  Output,
+  ViewChild,
+} from '@angular/core';
+import { FormControl, FormGroup, FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { MatButtonModule } from '@angular/material/button';
+import { MatButtonToggleModule } from '@angular/material/button-toggle';
+import { MatCardModule } from '@angular/material/card';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatInputModule } from '@angular/material/input';
+import { MatSelectModule } from '@angular/material/select';
+import { produce } from 'immer';
+
+import {
+  ColorMap,
+  DEFAULT_COLOR_MAP,
+  DEFAULT_SETTINGS,
+  MetaData,
+  MetadataSelectOption,
+  VisualizationSettings,
+} from '../../models/create-visualization-page-types';
+import { CellTypeDataService } from '../../services/service';
+import { MockCellTypeDataService } from '../../services/service.mock';
+
+@Component({
+  selector: 'cde-create-visualization-page',
+  standalone: true,
+  imports: [
+    CommonModule,
+    MatCardModule,
+    MatButtonToggleModule,
+    MatButtonModule,
+    MatFormFieldModule,
+    MatSelectModule,
+    MatInputModule,
+    FormsModule,
+    ReactiveFormsModule,
+  ],
+  providers: [
+    {
+      provide: CellTypeDataService,
+      useExisting: MockCellTypeDataService,
+    },
+  ],
+  templateUrl: './create-visualization-page.component.html',
+  styleUrl: './create-visualization-page.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CreateVisualizationPageComponent implements OnInit {
+  private readonly cellTypeDataService = inject(CellTypeDataService);
+
+  @Output() readonly visualize = new EventEmitter<VisualizationSettings>();
+
+  @ViewChild('fileInput') fileInput!: ElementRef<HTMLElement>;
+
+  defaultCellType = 'endothelial';
+
+  settings: VisualizationSettings = DEFAULT_SETTINGS;
+
+  colorMap: ColorMap = DEFAULT_COLOR_MAP;
+
+  visualizationForm = new FormGroup({
```

Comment:
Refer to https://angular.io/guide/reactive-forms#using-the-formbuilder-service-to-generate-controls to simplify the form control creation and add validation